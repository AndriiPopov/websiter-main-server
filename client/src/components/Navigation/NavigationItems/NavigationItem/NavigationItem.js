import React from 'react';
import { NavLink } from 'react-router-dom';

import classes from './NavigationItem.module.css';
import Svg from '../../../Svg/Svg';

const navigationItem = (props) => (
    <li className={classes.NavigationItem}>
        <NavLink 
            to={props.link}
            exact
            activeClassName={classes.Active} >
            <Svg icon={props.icon}/>
            {props.children}
        </NavLink>
    </li>
);

export default navigationItem;