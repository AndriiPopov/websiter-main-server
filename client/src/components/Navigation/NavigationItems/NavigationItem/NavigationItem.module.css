.NavigationItem {
    box-sizing: border-box;
    display: block;
    width: 100%;
}

.NavigationItem a {
    padding: 12px 0px;
    color: #555;
    text-decoration: none;
    width: 100%;
    box-sizing: border-box;
    display: flex;
    align-items: center;
    fill: #555;
}

.NavigationItem a svg {
    margin-right: 10px;
}

.NavigationItem a:hover,
.NavigationItem a:active,
.NavigationItem a.Active {
    color: #4084C8;
    fill: #4084C8;
}

.NavigationItem a {
    padding-left: 20px;
}