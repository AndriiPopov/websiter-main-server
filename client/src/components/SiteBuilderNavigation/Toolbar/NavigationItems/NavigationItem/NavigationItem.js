import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import classes from './NavigationItem.module.css';
import * as actions from '../../../../../store/actions/index';

export const NavigationItem = props => {
    const addedClasses = [ classes.NavigationItem ];
    if (props.activeItem === props.type) {
        addedClasses.push(classes.active);
    }

    return (
        <li 
            className={addedClasses.join(' ')}
            onClick={() => props.clickItem(props.type)}>
            {props.children}
        </li>
    );
};

NavigationItem.propTypes = {
    activeItem: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    clickItem: PropTypes.func
};

const mapStateToProps = state => {
    return {
        activeItem: state.topMenu.activeTopMenuItem
    };
};

const mapDispatchToProps = dispatch => {
    return {
        clickItem: (item) => dispatch(actions.setActiveTopMenuItem(item))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(NavigationItem);