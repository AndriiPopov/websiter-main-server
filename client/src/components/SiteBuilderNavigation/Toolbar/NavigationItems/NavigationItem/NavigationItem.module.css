.NavigationItem {
    color: #555;
    cursor: pointer;
    padding: 15px 20px;
    height: 100%;
    vertical-align: bottom;
    fill: #555;
    margin: 0px;
    display: flex;
    width: auto;
    align-items: center;
    box-sizing: border-box;
}

.NavigationItem:hover,
.NavigationItem:active,
.NavigationItem.active {
    background: #eee;
}