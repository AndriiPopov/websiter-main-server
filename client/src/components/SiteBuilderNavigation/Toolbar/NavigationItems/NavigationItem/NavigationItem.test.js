import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { NavigationItem } from './NavigationItem';

configure({adapter: new Adapter()});

describe('NavigationItem', () => {
    it('should be defined', () => {
        expect(NavigationItem).toBeDefined();
    });

    it('it should render children inside', () => {
        const tree = shallow(
            <NavigationItem
                activeItem="box"
                type="section" >
                <div><p>Test</p></div>
            </NavigationItem>
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('it should render children inside and have active class', () => {
        const tree = shallow(
            <NavigationItem
                activeItem="section"
                type="section" >
                <div><p>Test</p></div>
            </NavigationItem>
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});