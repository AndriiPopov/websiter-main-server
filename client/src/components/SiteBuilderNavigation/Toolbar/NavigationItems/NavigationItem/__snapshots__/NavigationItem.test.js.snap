// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem it should render children inside 1`] = `
<li
  className="NavigationItem"
  onClick={[Function]}
>
  <div>
    <p>
      Test
    </p>
  </div>
</li>
`;

exports[`NavigationItem it should render children inside and have active class 1`] = `
<li
  className="NavigationItem active"
  onClick={[Function]}
>
  <div>
    <p>
      Test
    </p>
  </div>
</li>
`;
